### ISSUE TITLE
`glide up` picks different versions on first vs second+ run after cache-clear

### ISSUE THREAD

**Issue by dt** [Masterminds/glide#706]
TL;DR: `glide cc && glide up && glide up` produces two different resolutions.

In attempting to update the dependencies of http://github.com/cockroachdb/cockroach, I ran `glide cc && glide up` to pick up a just-released version of an upstream lib.

In addition to the change I was expecting to see, I had many unrelated libs jump to very old versions. In the process of investigating, I re-ran `up` and saw them jump back to their current versions, and after a bit of pulling my hair out, I narrowed it down: The _first_ run on a clean cache produces a different resolution than any subsequent run. Checking our and re-clearing cache seems to reproduce the same first and subsequent resolutions, so it isn't just random.

Snipped output: https://gist.github.com/dt/f9ca4acb5bb27b9899d20c6f42bbf718
Full output with -debug: https://gist.github.com/dt/042eba2f391a89c9732e90d2ad5baa7c

**Comment by sdboyer** [comment]
if you've not seen it already, #592 may be related.

there are a handful of problems with the state management in the central cache that was introduced in `v0.12.0`. what you're describing here seems like a first, but it also may be a combination of these known issues.

i'm really swamped this week, but will look at the output you provided more when i get some time.

in the meantime...mmm, well, would you be willing to throw an hour at building glide from the `gps-integration` branch and seeing if these issues persist? idk if you'll run into some failure before you even get to this point, but if you don't, at minimum, you'll be able to specify the only deps you want to allow to update via a parameter to `glide update`.

**Comment by dt** [comment]
@sdboyer fwiw, this is distinctly *not* #592, but rather looks like a determinism issue uncovered by the workaround *for* #592 (clearing cache).

I'll see if I can give the gps branch a try later this week, thanks
